Web_Development/Javascript/Topics/5.Syntax/1.Data_Types/README.md

```diff
@@ -1 +1,103 @@
-# To Do
\ No newline at end of file
+# Data Types in JavaScript
+
+Just like any other programming language, JavaScript has data types too, however they might be different in their behaviour and properties unlike data types in other programming languages.
+
+## Primitive Types
+---
+JavaScript has 5 main primitive types :
+
+- String
+- Boolean
+- Number
+- Null and Undefined
+- BigInt
+
+*Note: Objects in JavaScript are a collection of properties and is not considered under Primitive types*
+
+## String Type
+---
+String type is used to represent textual data / information in JavaScript. It is a set of elements where each element occupies a position in the String. The first element is at 0th index, the next at index 1, and so on. The length of a String is the number of elements in it.
+Single quotes or double quotes may be used to assign a value to a string variable.
+
+Example:
+
+```js
+    let str = 'javascript';
+```
+
+*Note: In JavaScript Strings are immutable, that means once they are created we cannot modify the same*
+
+## Boolean Type
+---
+Boolean type in JavaScript represents two values, logical true (or 1) or logical false (or 0)
+
+Example:
+
+```js
+    let isGameOver = true;
+    isGameOver = 0;
+```
+
+## Number Type
+---
+JavaScript has only one number type. Decimals, Integers, Whole numbers all are represented using a single type called Number. Unlike, other programming languages, JavaScript doesn't have different data types for different types of numbers.
+
+Example:
+
+```js
+    let x = 12.34;
+    let y = 123;
+    let negativeNumber = -34;
+```
+
+## Null Type
+---
+The Null type represents the intentional absence of a value in Javascript. It is a primitive type and is treated as falsy for boolean operations.
+
+Example:
+
+```js
+    let value = null;
+```
+
+## Undefined Type
+---
+The Undefined type in JavaScript is its way of saying, it doesn't know what's present inside something or if something is not defined. For example a variable that has not been assigned a value is of type undefined.
+
+Example:
+
+``` js
+    let x; // x is not assigned any value
+    typeof(x); // hence, it's type is undefined
+```
+
```

Comment:
Excellent work! Just get rid of the extra indentation in code blocks and you're good to be merged!